x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -38,53 +38,43 @@ fn get_value_internal(context: &Context<StateKeys>, of: Address) -> Count {
 
 #[cfg(test)]
 mod tests {
-    use simulator::{Endpoint, Step, TestContext};
+    use simulator::{build_simulator, TestContext};
     use wasmlanche_sdk::Address;
 
     const PROGRAM_PATH: &str = env!("PROGRAM_PATH");
 
     #[test]
     fn init_program() {
-        let mut simulator = simulator::ClientBuilder::new().try_build().unwrap();
+        let mut simulator = build_simulator().unwrap();
 
-        simulator
-            .run_step(&Step::create_program(PROGRAM_PATH))
-            .unwrap();
+        simulator.create_program(PROGRAM_PATH).unwrap();
     }
 
     #[test]
     fn increment() {
-        let mut simulator = simulator::ClientBuilder::new().try_build().unwrap();
+        let mut simulator = build_simulator().unwrap();
```

Comment:
You could probably change this to be `Simulator::new()` and use `expect` *inside* the `new` function. I can't imagine how anyone would try to recover from the simulator instantiation erroring out

Reply:
If we have a `Simulator::new()` function the caller would need to explicitly specify the generic type arguments, I think it's cleaner to just have a `build_simulator` function and have the caller worry about this

something like `let simulator = Simulator::<Childstdin, ...>::new()`